### PR TITLE
Fixed #30588 -- Fixed crash of autoreloader when __main__ module doesn't have __file__ attribute.

### PR DESCRIPTION
**Pull request by orf**
Ticket: https://code.djangoproject.com/ticket/30588#ticket

### CODE REVIEW

File: docs/spelling_wordlist [hunk 1]
```diff
@@ -296,6 +296,7 @@ ip
 ipsum
 IPv
 IPython
+ipdb
```
Comment (MarkusH) [inline docs/spelling_wordlist]: Could you sort this above "ipsum" on line 296, please.

Comment (orf) [reply docs/spelling_wordlist]: Done 👍

**Comment by MarkusH** [comment]
buildbot, test this please

File: docs/releases/2.2.3.txt [hunk 2]
```diff
@@ -12,3 +12,6 @@ Bugfixes
 * Fixed a regression in Django 2.2 where :class:`~django.db.models.Avg`,
   :class:`~django.db.models.StdDev`, and :class:`~django.db.models.Variance`
   crash with ``filter`` argument (:ticket:`30542`).
+
+* Fixed a regression in Django 2.2 where using ipdb would prevent the
+  auto-reloader from working (:ticket:`30588`).
```
Comment (felixxm) [inline docs/releases/2.2.3.txt]: Maybe:
```
* Fixed a regression in Django 2.2.2 where auto-reloader crashes with
  ``AttributeError``, e.g. when using ``ipdb`` (:ticket:`30588`).
```
``ipdb`` can be removed from ``docs/spelling_wordlist``.

File: django/utils/autoreload.py [hunk 3]
```diff
@@ -117,6 +117,10 @@ def iter_modules_and_files(modules, extra_files):
         if not isinstance(module, ModuleType):
             continue
         if module.__name__ == '__main__':
+            # In some cases __main__.__file__ will not be set, such as while running
+            # in an ipdb shell.
+            if not getattr(module, '__file__', None):
+                continue
```
[the remark this answers is not in the document]
Comment (orf) [reply django/utils/autoreload.py]: Sure, good idea

File: docs/releases/2.2.3.txt (hunk 2 above)
Comment (orf) [reply docs/releases/2.2.3.txt]: Done 👍

**Comment by felixxm** [comment]
@orf Thanks!

**Pull request merged**